src/cascadia/QueryExtension/ExtensionPalette.cpp

```diff
@@ -489,4 +436,54 @@ namespace winrt::Microsoft::Terminal::Query::Extension::implementation
         // Clear the text box each time we close the dialog. This is consistent with VsCode.
         _queryBox().Text(winrt::hstring{});
     }
+
+    ChatMessage::ChatMessage(winrt::hstring content, bool isQuery) :
+        _messageContent{ content },
+        _isQuery{ isQuery }
+    {
+        _richBlock = Microsoft::Terminal::UI::Markdown::Builder::Convert(_messageContent, L"");
+        const auto resources = Application::Current().Resources();
+        if (_isQuery)
+        {
+            if (const auto textBrush = resources.Lookup(box_value(L"TextOnAccentFillColorPrimaryBrush")).try_as<Windows::UI::Xaml::Media::SolidColorBrush>())
+            {
+                _richBlock.Foreground(textBrush);
+            }
+        }
+        else
+        {
+            for (const auto& b : _richBlock.Blocks())
+            {
+                if (const auto& p{ b.try_as<Windows::UI::Xaml::Documents::Paragraph>() })
+                {
+                    for (const auto& line : p.Inlines())
+                    {
+                        if (const auto& otherContent{ line.try_as<Windows::UI::Xaml::Documents::InlineUIContainer>() })
+                        {
+                            if (const auto& codeBlock{ otherContent.Child().try_as<Microsoft::Terminal::UI::Markdown::CodeBlock>() })
+                            {
+                                codeBlock.Margin({ 0, 8, 0, 8 });
+                                codeBlock.PlayButtonVisibility(Windows::UI::Xaml::Visibility::Visible);
+                                if (const auto backgroundBrush = resources.Lookup(box_value(L"ControlAltFillColorSecondaryBrush")).try_as<Windows::UI::Xaml::Media::SolidColorBrush>())
```

Comment:
Does a lookup really need boxing? I thought the keys were strings?

Reply:
`Lookup` takes an `IInspectable`!